src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -21,26 +21,7 @@ public class DefaultTraceListener : TraceListener
     {
         private class DefaultTraceDebugProvider : DebugProvider
```

Comment:
So all this really does now is to let you call the `WriteCore` method with a bunch of extra overhead. Is that correct? Would it be better to just call it directly without the ceremony?

Reply:
Exactly. Now with the duplicated code removed I can just easily make WriteCore the override-able Write method of DebugProvider.

Then the Debug indentation logic will go out of DebugProvider and back to Debug class instead.

Next commit will be removing the delegate.

Reply:
In the new PRs, I removed this class, and just called WriteCore through DebugProvider.

Please note, in this PR and in the previous one I use DebugProvider to wire Debug with Trace Listeners.
But in the newer PRs where I finalized the new class names and APIs:
- I named the class responsible for linking Trace with Debug `DebugInternal`.
- I used the name DebugProvider to refer to the class we indent to make public to help users override ShowDialog and WriteCore.